app/services/idv/phone_step.rb

```diff
@@ -23,6 +16,38 @@ def failure_reason
       return :warning if idv_result[:success] != true
     end
 
+    def async_state
+      dcs_uuid = idv_session.idv_phone_step_document_capture_session_uuid
+      dcs = DocumentCaptureSession.find_by(uuid: dcs_uuid)
+      return ProofingDocumentCaptureSessionResult.none if dcs_uuid.nil?
+      return ProofingDocumentCaptureSessionResult.timed_out if dcs.nil?
+
+      proofing_job_result = dcs.load_proofing_result
+      return ProofingDocumentCaptureSessionResult.timed_out if proofing_job_result.nil?
+
+      if proofing_job_result.result
+        proofing_job_result.done!
+
+        proofing_job_result
+      elsif proofing_job_result.pii
+        ProofingDocumentCaptureSessionResult.in_progress
+      end
+    end
+
+    def async_state_done(async_state)
+      @idv_result = async_state.result
+      pii = async_state.pii
+
+      increment_attempts_count unless failed_due_to_timeout_or_exception?
+      success = idv_result[:success]
+      handle_successful_proofing_attempt(pii) if success
+      idv_session.idv_phone_step_document_capture_session_uuid = nil
+      FormResponse.new(
+        success: success, errors: idv_result[:errors],
+        extra: extra_analytics_attributes
+      )
```

Comment:
Is `idv_result` a plain hash here? Or a FormResponse object? We added `FormResponse#merge` so we can simplify combining/adding data here.

```
idv_result = FormResponse.new(**idv_result) # or maybe add .from_hash if needed?
idv_result.merge(FormResponse.new(success: true, extra: extra_analytics_attributes)
```

Inside it has `success: self.success && other.success` so it should be fine

Reply:
It is, but I don't mind the explicitness. I don't know if this is the ideal form, but I like that the transformation gives some insight into what both of them are holding.